sos/report/plugins/snap.py

```diff
@@ -23,8 +24,42 @@ def setup(self):
         self.add_cmd_output("snap list --all", root_symlink="installed-snaps")
         self.add_cmd_output([
             "snap --version",
-            "snap changes"
+            "snap version",
+            "snap whoami",
+            "snap model --verbose",
+            "snap model --serial --verbose",
+            "snap services",
+            "snap connections",
+            "snap changes",
+            "snap changes --abs-time",
+            "snap validate",
+            "snap debug state --abs-time --changes /var/lib/snapd/state.json",
+            "snap debug stacktraces",
+            "snap get system -d",
         ])
+
+        all_pkgs = self.policy.package_manager.packages
+
+        for pkg_name in all_pkgs:
+            pkg = self.policy.package_manager.pkg_by_name(pkg_name)
+            if pkg['pkg_manager'] == 'snap':
+                self.add_cmd_output(f"snap connections {pkg['name']}")
+
         self.add_cmd_output("snap debug connectivity", timeout=10)
 
+        snap_changes = sos_get_command_output('snap changes')
```

Comment:
Use `Plugin.exec_cmd()` here, rather than directly calling the utilities method.